### PR TITLE
[4.0] Normalise Featured Contact List

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes
Follow up on https://github.com/joomla/joomla-cms/pull/30813 and https://github.com/joomla/joomla-cms/pull/30859
Warning: these 2 PR have to be present in your test site to test the Clear button.

**npm ci has to be applied** 

Changes:
Normalises the table display
Adds a Title Search Filter, Filter and Clear buttons.
a11y legend and scope added
Modified the search placeholder (it is also used in Contact List but I forgot to modify it there)
Used Title instead of Name for column heading, as in Contact list

### Testing Instructions
Create some contacts and set them to featured.
Create a Featured Contacts menu item.
Display in frontend.

After patch, test that the Filter and the Clear button work as should

### Actual result BEFORE applying this Pull Request

Screen Shot 2020-10-02 at 11 50 02

### Expected result AFTER applying this Pull Request

Screen Shot 2020-10-02 at 12 14 04

### NOTE
No idea why we display the NUM in the first column. 
If nobody knows why, we can get rid of it.

Meaanwhile, I kept in the code
```
					<tr class="<?php echo ($i % 2) ? 'odd' : 'even'; ?>" itemscope itemtype="https://schema.org/Person">
						<td class="item-num">
							<?php echo $i; ?>
						</td>
```

### CODE REVIEW

**Comment by chmst** [comment]
I'd like to get rid of the Number!

**Comment by infograf768** [comment]
> I'd like to get rid of the Number!

Me too, if it is BC.

**Comment by brianteeman** [comment]
There is no b/c issue

**Comment by chmst** [comment]
Tested with success. If you remove the number column, I'll re-test. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30868">issues.joomla.org/tracker/joomla-cms/30868</a>.</sub>

**Comment by infograf768** [comment]
> There is no b/c issue

@brianteeman 
Do you mean I can get rid of that column?

Any idea about this very specific `<tr>`
`<tr class="<?php echo ($i % 2) ? 'odd' : 'even'; ?>" itemscope itemtype="https://schema.org/Person">`

Should I replace it with the same stuff we have in the category list?
```
						<?php if ($this->items[$i]->published == 0) : ?>
							<tr class="system-unpublished featured-list-row<?php echo $i % 2; ?>">
						<?php else : ?>
							<tr class="featured-list-row<?php echo $i % 2; ?>" >
						<?php endif; ?>
```

**Comment by brianteeman** [comment]
1. yes - its just a display you are not removing data

2. that css is/was  to create "stripes"

**Comment by infograf768** [comment]
@chmst @brianteeman 
please test again 
also moved Unpublished badge below contact title (log to see)
Screen Shot 2020-10-03 at 09 19 16

File: components/com_contact/tmpl/featured/default_items.php [hunk 1]
```diff
@@ -15,162 +15,195 @@
...
 ?>
-
 <div class="com-contact-featured__items">
-	<?php if (empty($this->items)) : ?>
-		<p class="com-contact-featured__message"> <?php echo Text::_('COM_CONTACT_NO_CONTACTS'); ?>	 </p>
-	<?php else : ?>
-
 	<form action="<?php echo htmlspecialchars(Uri::getInstance()->toString()); ?>" method="post" name="adminForm" id="adminForm">
+		<?php if ($this->params->get('filter_field') || $this->params->get('show_pagination_limit')) : ?>
 		<fieldset class="com-contact-featured__filters filters">
```
Comment (chmst) [inline components/com_contact/tmpl/featured/default_items.php]: Filter and list limit are not a fieldset. They are not related to each other.

File: components/com_contact/tmpl/featured/default_items.php [hunk 2]
```diff
@@ -15,162 +15,195 @@
...
-		<?php if ($this->params->get('show_pagination_limit')) : ?>
-			<div class="display-limit">
-				<?php echo Text::_('JGLOBAL_DISPLAY_NUM'); ?>&#160;
-				<?php echo $this->pagination->getLimitBox(); ?>
-			</div>
-		<?php endif; ?>
-		<input type="hidden" name="filter_order" value="<?php echo $listOrder; ?>">
-			<input type="hidden" name="filter_order_Dir" value="<?php echo $listDirn; ?>">
-		</fieldset>
+			<legend class="sr-only"><?php echo Text::_('COM_CONTACT_FORM_FILTER_LEGEND'); ?></legend>
```
Comment (chmst) [inline components/com_contact/tmpl/featured/default_items.php]: no need for a legend

**Comment by chmst** [comment]
The list is a big improvement. 

For the filter, I have some doubts. 
- The form is only the filter, not the whole list
- Search field and list limit are no fieldset, they don't belong together
- The clear button does not clear the input.

**Comment by infograf768** [comment]
The clear button works if
> Follow up on #30813 and #30859
> Warning: these 2 PR have to be present in your test site to test the Clear button.

> npm ci has to be applied

File: components/com_contact/tmpl/featured/default_items.php (hunk 2 above)
Comment (infograf768) [reply components/com_contact/tmpl/featured/default_items.php]: fieldset needs a legend (and wave asks for it): see 

https://github.com/joomla/joomla-cms/blob/06fe4d226bf0d70f2ddf7cd5a213704089349a7b/components/com_content/tmpl/category/default_articles.php#L69-L70

https://github.com/joomla/joomla-cms/blob/06fe4d226bf0d70f2ddf7cd5a213704089349a7b/components/com_contact/tmpl/category/default_items.php#L35-L36

File: components/com_contact/tmpl/featured/default_items.php (hunk 1 above)
Comment (infograf768) [reply components/com_contact/tmpl/featured/default_items.php]: Please look at similar code for 
components/com_content/tmpl/category/default_articles.php

**Comment by infograf768** [comment]
@chmst
I replied to your concerns above, I guess.

**Comment by brianteeman** [comment]
The point was that it's not a fiedset

**Comment by infograf768** [comment]
> The point was that it's not a fiedset

Then you may have to change the code for articles in 3.x and for articles and contacts lists in 4.0-dev, which you tested OK

https://github.com/joomla/joomla-cms/blob/staging/components/com_content/views/category/tmpl/default_articles.php#L67-L104

**Comment by brianteeman** [comment]
I have no opinion on if it should be a fieldset or not. I was merely explaining why @chmst said that you should not have the legend.

**Comment by infograf768** [comment]
Animated gif concerning the Clear filter.
limitbox_contacts-list-animated

**Comment by infograf768** [comment]
I obviously can take off the part concerning fieldset, here and in List Contacts in a Category, if desired.

**Comment by infograf768** [comment]
I am asking @zwiastunsw to please help on this decision as we have only one input (the searchfield)
One aspect is sure: if we keep the fieldset, it must have a legend.

**Comment by infograf768** [comment]
After discussing with @chmst and @zwiastunsw fieldset is not necessary.
Reworked the tmpl.

Please test.

### Note
Another PR will follow for contacts list and articles list.

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 510dc5f617976464d48371cf3f30bcfe9dc321d8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30868">issues.joomla.org/tracker/joomla-cms/30868</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 510dc5f617976464d48371cf3f30bcfe9dc321d8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30868">issues.joomla.org/tracker/joomla-cms/30868</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30868">issues.joomla.org/tracker/joomla-cms/30868</a>.</sub>

File: components/com_contact/tmpl/featured/default_items.php [hunk 3]
```diff
@@ -15,162 +15,188 @@
...
+				<label class="filter-search-lbl sr-only" for="filter-search">
+					<?php echo Text::_('COM_CONTACT_FILTER_SEARCH_DESC'); ?>
+				</label>
+				<input
+					type="text"
+					name="filter-search"
+					id="filter-search"
+					value="<?php echo $this->escape($this->state->get('list.filter')); ?>"
+					class="inputbox" onchange="document.adminForm.submit();"
+					title="<?php echo Text::_('COM_CONTACT_FILTER_SEARCH_DESC'); ?>"
```
Comment (Quy) [inline components/com_contact/tmpl/featured/default_items.php]: Probably don't need title attribute when identical to placeholder attribute?

Comment (infograf768) [reply components/com_contact/tmpl/featured/default_items.php]: No idea. Did not touch that part. Tags have both, newsfeeds not... Should aria-label be used instead?
Ready to modify if necessary.

Comment (HLeithner) [reply components/com_contact/tmpl/featured/default_items.php]: based on the [Specs](https://www.w3.org/TR/html-aam-1.0/#input-type-text-input-type-password-input-type-search-input-type-tel-input-type-email-input-type-url-and-textarea-element-accessible-name-computation) you could remove the title.

**Comment by infograf768** [comment]
title removed as requested.
I guess no need to test again.

File: components/com_contact/tmpl/featured/default_items.php (hunk 3 above)
Comment (infograf768) [reply components/com_contact/tmpl/featured/default_items.php]: done

**Comment by HLeithner** [comment]
Thanks

**Comment by infograf768** [comment]
Similar PR for articles
https://github.com/joomla/joomla-cms/pull/30981

**Pull request merged**